app.ts

```diff
@@ -0,0 +1,135 @@
+let recordCount: number, startIndex: number, endIndex: number;
+let columnHeaders: any[];
+let resizeTimer: number , rowsPerView : number;
+
+window.onload = () => getRecordCount();
+ 
+// calculating how many records to display based on widow size
+function calRowPerView() {
+  let x = Math.floor((window.innerHeight - 245) / 34);
+  
+  if (x < 0) {
+    alert("Window screen to small to display records , Resize window");
+  }
+  return x;
+}
+
+function loadGrid(start: number) {
+  rowsPerView = calRowPerView();
```

Comment:
Here, rowsPerView is assigned a value every time data is loaded, it needs to be assigned only on startup or when the window is resized.